SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -19,13 +19,123 @@ DEFINE_LOG_CATEGORY(LogInterestFactory);
 
 namespace SpatialGDK
 {
+struct FrequencyConstraint
+{
+	float Frequency;
+	SpatialGDK::QueryConstraint Constraint;
+};
+static TArray<FrequencyConstraint> CheckoutConstraints;
+
 // The checkout radius constraint is built once for all actors in CreateCheckoutRadiusConstraint as it is equivalent for all actors.
 // It is built once per net driver initialisation.
 static QueryConstraint ClientCheckoutRadiusConstraint;
 
 // Cache the result type of client Interest queries.
 static TArray<Worker_ComponentId> ClientResultType;
 
+QueryConstraint CreateLegacyNetCullDistanceConstraint(USpatialClassInfoManager* ClassInfoManager)
+{
+	QueryConstraint CheckoutRadiusConstraint;
+
+	CheckoutRadiusConstraint.OrConstraint.Add(CheckoutRadiusConstraintUtils::GetDefaultCheckoutRadiusConstraint());
+
+	// Get interest distances for each actor.
+	TMap<UClass*, float> ActorComponentSetToRadius = CheckoutRadiusConstraintUtils::GetActorTypeToRadius();
+
+	// For every interest distance that we still want, build a map from radius to list of actor type components that match that radius.
+	TMap<float, TArray<UClass*>> DistanceToActorTypeComponents = CheckoutRadiusConstraintUtils::DedupeDistancesAcrossActorTypes(
+		ActorComponentSetToRadius);
+
+	// The previously built map dedupes spatial constraints. Now the actual query constraints can be built of the form:
+	// OR(AND(cyl(radius), OR(actor 1 components, actor 2 components, ...)), ...)
+	// which is equivalent to having a separate spatial query for each actor type if the radius is the same.
+	TArray<QueryConstraint> CheckoutRadiusConstraints = CheckoutRadiusConstraintUtils::BuildNonDefaultActorCheckoutConstraints(
+		DistanceToActorTypeComponents, ClassInfoManager);
+
+	// Add all the different actor queries to the overall checkout constraint.
+	for (auto& ActorCheckoutConstraint : CheckoutRadiusConstraints)
+	{
+		CheckoutRadiusConstraint.OrConstraint.Add(ActorCheckoutConstraint);
+	}
+
+	return CheckoutRadiusConstraint;
+}
+
+QueryConstraint CreateNetCullDistanceConstraint(USpatialClassInfoManager* ClassInfoManager)
+{
+	QueryConstraint CheckoutRadiusConstraintRoot;
+
+	const TArray<float> NetCullDistances = ClassInfoManager->GetNetCullDistances();
+
+	for (float NetCullDistance : NetCullDistances)
```

Comment:
Why are we not just iterating the pairs? It seems like we get the keys and then immediately look them up in the map.

Reply:
Smart